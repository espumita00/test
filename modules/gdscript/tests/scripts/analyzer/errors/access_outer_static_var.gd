# This works for outer constants, enums, and classes. We _could_ implement this for static variables,
# but it **SHOULD NOT** be done since inner classes are initialized first. See GH-82141.

static var outer_static_var: int

class InnerClass:
	static func _static_init() -> void:
		print(outer_static_var)

func test():
	pass
