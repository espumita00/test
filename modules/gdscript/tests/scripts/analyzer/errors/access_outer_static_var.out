GDTEST_ANALYZER_ERROR
Identifier "outer_static_var" not declared in the current scope.
